proj4/src/main/java/org/osgeo/proj4j/datum/Datum.java

```diff
@@ -193,14 +193,9 @@ public boolean isEqual(Datum datum)
           return false;
       }
       return true;
-    } 
-    /* 
-     //TODO: complete
-    else if( this.datum_type == Proj4js.common.PJD_GRIDSHIFT ) {
-      return strcmp( pj_param(this.params,"snadgrids").s,
-                     pj_param(dest.params,"snadgrids").s ) == 0;
+    } else if (getTransformType() == TYPE_GRIDSHIFT) {
+        return grids.equals(datum.grids);
```

Comment:
Should make sure that Grid and ConversionTable have structural comparison methods.